113 eurostat

Juan-Mateos opened the pull request:
Partially closes #113 

Some notes:

* All regions in France below NUTS0 are missing 🇫🇷 
* NUTS0 codes = ISO2 codes I think.
* The nuts0 table includes countries outside of the EU e.g. China, US
* We need to change the framework categories. These variables are temporarily labelled as "knowledge exchange"
* I have created a couple of text files for virtual environment installs. These should be replaced with cookiecutter requirements when we move to that structure

Commit:
added script to make indicators

Commit:
added script to make indicators

Commit:
added r&d expenditure indicators

Commit:
added r&d expenditure indicators

Commit:
change rd file names

Commit:
change rd file names

Commit:
added indicators and yaml files

Commit:
added indicators and yaml files

mindrones commented on ds/data/processed/eurostat/rd_government_eur_hab.nuts0.yaml:
```diff
@@ -0,0 +1,29 @@
+  nuts_year_spec:
+    type: NutsRegion.nuts_year_spec
+  value:
+    data_type: int
```
(1) the data seem to be `float` with 1 digit `format`

mindrones commented on ds/data/processed/eurostat/rd_government_eur_hab.nuts1.yaml:
```diff
@@ -0,0 +1,29 @@
+  nuts_year_spec:
+    type: NutsRegion.nuts_year_spec
+  value:
+    data_type: int
```
See (1)

mindrones commented on ds/data/processed/eurostat/rd_government_eur_hab.nuts1.csv:
```diff
@@ -0,0 +1,438 @@
+year,nuts_id,nuts_year_spec,rd_government_eur_hab
+2015,AT1,2016,
+2017,AT1,2016,
```
(2) the value seems to be missing here

Juan-Mateos replied:
Data are missing at source

mindrones replied:
In that case shouldn't we have a default like zero?

mindrones commented on ds/data/processed/eurostat/rd_government_eur_hab.nuts2.csv:
```diff
@@ -0,0 +1,1227 @@
+year,nuts_id,nuts_year_spec,rd_government_eur_hab
+2015,AT11,2016,
+2017,AT11,2016,
```
See (2)

Juan-Mateos replied:
Data are missing at source

mindrones commented on ds/data/processed/eurostat/rd_government_eur_hab.nuts2.yaml:
```diff
@@ -0,0 +1,29 @@
+  nuts_year_spec:
+    type: NutsRegion.nuts_year_spec
+  value:
+    data_type: int
```
See (1)

mindrones commented on ds/data/processed/eurostat/rd_private_eur_hab.nuts0.yaml:
```diff
@@ -0,0 +1,29 @@
+  nuts_year_spec:
+    type: NutsRegion.nuts_year_spec
+  value:
+    data_type: int
```
See (1)

mindrones commented on ds/data/processed/eurostat/rd_private_eur_hab.nuts1.csv:
```diff
@@ -0,0 +1,389 @@
+2016,MT0,2016,80.7
+2017,MT0,2016,93.6
+2018,MT0,2016,98.8
+2014,NL1,2016,
+2015,NL1,2016,
+2016,NL1,2016,
+2017,NL1,2016,
+2018,NL1,2016,
+2014,NL2,2016,
+2015,NL2,2016,
+2016,NL2,2016,
+2017,NL2,2016,
+2018,NL2,2016,
+2014,NL3,2016,
+2015,NL3,2016,
+2016,NL3,2016,
+2017,NL3,2016,
+2018,NL3,2016,
+2014,NL4,2016,
+2015,NL4,2016,
+2016,NL4,2016,
+2017,NL4,2016,
+2018,NL4,2016,
```
See (2)

Juan-Mateos replied:
Data are missing at source

mindrones commented on ds/data/processed/eurostat/rd_private_eur_hab.nuts1.yaml:
```diff
@@ -0,0 +1,29 @@
+  nuts_year_spec:
+    type: NutsRegion.nuts_year_spec
+  value:
+    data_type: int
```
See (1)

mindrones commented on ds/data/processed/eurostat/rd_private_eur_hab.nuts2.csv:
```diff
@@ -0,0 +1,1120 @@
+2015,BE10,2016,629.8
+2016,BE10,2016,723.4
+2017,BE10,2016,840.3
+2014,BE21,2016,
+2015,BE21,2016,
+2016,BE21,2016,
+2017,BE21,2016,
+2014,BE22,2016,
+2015,BE22,2016,
+2016,BE22,2016,
```
See (2)

Juan-Mateos replied:
Data are missing at source

mindrones commented on ds/data/processed/eurostat/rd_private_eur_hab.nuts2.csv:
```diff
@@ -0,0 +1,1120 @@
+2018,ES62,2016,98.3
+2014,ES63,2016,
+2015,ES63,2016,0.0
+2016,ES63,2016,
+2017,ES63,2016,
+2018,ES63,2016,
+2014,ES64,2016,
```
See (2)

Juan-Mateos replied:
Data are missing at source

mindrones commented on ds/data/processed/eurostat/rd_private_eur_hab.nuts2.yaml:
```diff
@@ -0,0 +1,29 @@
+  nuts_year_spec:
+    type: NutsRegion.nuts_year_spec
+  value:
+    data_type: int
```
See (1)

mindrones reviewed (changes requested):
First pass:
- I've left comments below
- The [new structure](https://github.com/nestauk/eurito_indicators/pull/121) requires having processed data in `ds/outputs/data/processed/` rather than in `ds/data/processed/`

mindrones commented on ds/data/processed/eurostat/rd_government_eur_hab.nuts0.yaml:
```diff
@@ -0,0 +1,29 @@
+title: R&D Expenditure (Government)
+subtitle: R&D expenditure per capita from government sources
+endpoint_url: https://ec.europa.eu/eurostat/api/dissemination/sdmx/2.1/data/RD_E_GERDREG$DEFAULTVIEW/
+framework_group: knowledge_exchange
```
(3) For now I'd change this to `EUROSTAT` and in `ds/outputs/data/aux/framework.json` I'd add a new section like:

```
{
    "description": "EUROSTAT indicators",
    "id": "EUROSTAT",
    "label": "EUROSTAT",
    "order": 0
}
```

so that we would have a section with all EUROSTAT indicators in their own section.

mindrones commented on ds/data/processed/eurostat/rd_government_eur_hab.nuts1.yaml:
```diff
@@ -0,0 +1,29 @@
+title: R&D Expenditure (Government)
+subtitle: R&D expenditure per capita from government sources
+endpoint_url: https://ec.europa.eu/eurostat/api/dissemination/sdmx/2.1/data/RD_E_GERDREG$DEFAULTVIEW/
+framework_group: knowledge_exchange
```
See (3)

mindrones commented on ds/data/processed/eurostat/rd_government_eur_hab.nuts2.yaml:
```diff
@@ -0,0 +1,29 @@
+title: R&D Expenditure (Government)
+subtitle: R&D expenditure per capita from government sources
+endpoint_url: https://ec.europa.eu/eurostat/api/dissemination/sdmx/2.1/data/RD_E_GERDREG$DEFAULTVIEW/
+framework_group: knowledge_exchange
```
See (3)

mindrones commented on ds/data/processed/eurostat/rd_private_eur_hab.nuts0.yaml:
```diff
@@ -0,0 +1,29 @@
+title: R&D Expenditure (Private)
+subtitle: R&D expenditure per capita from private sector sources
+endpoint_url: https://ec.europa.eu/eurostat/api/dissemination/sdmx/2.1/data/RD_E_GERDREG$DEFAULTVIEW/
+framework_group: knowledge_exchange
```
See (3)

mindrones commented on ds/data/processed/eurostat/rd_private_eur_hab.nuts1.yaml:
```diff
@@ -0,0 +1,29 @@
+title: R&D Expenditure (Private)
+subtitle: R&D expenditure per capita from private sector sources
+endpoint_url: https://ec.europa.eu/eurostat/api/dissemination/sdmx/2.1/data/RD_E_GERDREG$DEFAULTVIEW/
+framework_group: knowledge_exchange
```
See (3)

mindrones commented on ds/data/processed/eurostat/rd_private_eur_hab.nuts2.yaml:
```diff
@@ -0,0 +1,29 @@
+title: R&D Expenditure (Private)
+subtitle: R&D expenditure per capita from private sector sources
+endpoint_url: https://ec.europa.eu/eurostat/api/dissemination/sdmx/2.1/data/RD_E_GERDREG$DEFAULTVIEW/
+framework_group: knowledge_exchange
```
See (3)

mindrones reviewed (commented):
Added more comments regarding the indicators framework

Commit:
addressing pr comments

Commit:
addressing pr comments

Juan-Mateos commented:
I changed `data_types` to float (sorry about that!) and `framework_group` to EUROSTAT.

I haven't added the new section to `framework.json`.

All missing data are unavailable from eurostat. Many EU countries seem to be quite erratic in their provision of sub-national data.

Commit:
change target directory in indicator production script

Commit:
change target directory in indicator production script

mindrones commented:
@Juan-Mateos re: 2 (missing data), can we filter those rows out?

Commit:
removed empty rows from tables

Commit:
removed empty rows from tables

Juan-Mateos commented:
I have:
* Written a script that collects, cleans and processes all Eurostat science and technology regional indicators
  * The script also populates automatically part of the schema (although not all - it is difficult to do it automatically and manually is not an option because we have hundreds of indicators (see below)) 
* Updated the requirements
* Ran the script and saved indicators and schema
* Removed the older test indicators that were produced using the old schema
* Linted and documented all the code

Some issues:
* We currently don't have automatic detection of string formatting
* We are currently not populating some fields in the schema and some of the fields of the schema are perhaps too long
* The names of the indicators, based on eurostat ids for data sources and variables are not very readable.
* There are ~200 indicators! As many as there are units / sectoral categories within the Eurostat tables
* Most of the indicators are available at NUTS0, NUTS1 and NUTS2 level but not NUTS3.

Commit:
solved requirements conflict

Commit:
solved requirements conflict

Commit:
fixed conflict

Commit:
fixed conflict

Commit:
rebased to dev and fixed conflicts, fixed bugs in script, updated csvs and json

Commit:
rebased to dev and fixed conflicts, fixed bugs in script, updated csvs and json

Commit:
add schema with fixed typos

Commit:
add schema with fixed typos

mindrones reviewed (changes requested):
First pass:
- there are a number of files ending in ` 2.csv` that should be removed as they are duplicates and don't have a correspondent schema file [1]
- a number of schema files are not indented, which would make reviewing them difficult (e.g. `ds/outputs/data/processed/eurostat/hrst_st_rcat_hrst_pc_act.json`)

[1]

```
ds/outputs/data/processed/eurostat/hrst_st_rcat_hrst_pc_act 2.csv
ds/outputs/data/processed/eurostat/hrst_st_rcat_hrstc_pc_act 2.csv
ds/outputs/data/processed/eurostat/hrst_st_rcat_hrstc_pc_pop 2.csv
ds/outputs/data/processed/eurostat/hrst_st_rcat_hrste_pc_act 2.csv
ds/outputs/data/processed/eurostat/hrst_st_rcat_hrsto_pc_act 2.csv
ds/outputs/data/processed/eurostat/hrst_st_rcat_hrsto_ths 2.csv
ds/outputs/data/processed/eurostat/hrst_st_rcat_se_pc_act 2.csv
```

Commit:
removed duplicated indicators and indented schema

Commit:
removed duplicated indicators and indented schema

Commit:
remove eurostat folder

Commit:
remove eurostat folder

Commit:
removed duplicated indicators etc

Commit:
removed duplicated indicators etc

mindrones reviewed (changes requested):
Some changes required:

1) we need to add [1] to `ds/outputs/data/aux/framework.json`

2) we need to change `"framework_group": "Eurostat"` to `"framework_group": "EUROSTAT"` in all schema files

3) a number of indicators have long `title` and `subtitle` [3]

4) I think there might be some duplicate indicator `id` as for some of them the sidebar does not scroll to them, will investigate while you're fixing the above

5) there are a number of datapoints like

   ```
   1988,NUTS,2010,FR101,3,6.25
   ```

   where:
   
   1. I'm not sure why `1988` would be associated with `2010` and not with `2003` (the first official release of NUTS)
   
   2. most importantly, the association `region_year_spec`/`region_id` does not exist in the NUTS spec: we support [these](https://raw.githubusercontent.com/nestauk/svizzle_atlas_distro/79668ec058da88580c5b456519f6a701afab9e7b/NUTS/unifiedNuts.json)

---

[1]

```
{
	"description": "EUROSTAT indicators",
	"id": "EUROSTAT",
	"label": "EUROSTAT",
	"order": 1
}
```

[3]

```
-- id: ai_covid_count
 |- subtitle: 164
-- id: airborne_count
 |- subtitle: 152
-- id: airborne_lq
 |- subtitle: 240
-- id: antibodies_count
 |- subtitle: 154
-- id: antibodies_lq
 |- subtitle: 242
-- id: antiviral_drugs_count
 |- subtitle: 159
-- id: antiviral_drugs_lq
 |- subtitle: 247
-- id: artificial_intelligence_count
 |- subtitle: 160
-- id: deep_learning_count
 |- subtitle: 150
-- id: detection_diagnosis_count
 |- subtitle: 163
-- id: detection_diagnosis_lq
 |- subtitle: 251
-- id: detection_sequencing_count
 |- subtitle: 164
-- id: detection_sequencing_lq
 |- subtitle: 252
-- id: education_count
 |- subtitle: 153
-- id: education_lq
 |- subtitle: 241
-- id: genetic_analysis_count
 |- subtitle: 160
-- id: genetic_analysis_lq
 |- subtitle: 248
-- id: healthcare_count
 |- subtitle: 154
-- id: healthcare_lq
 |- subtitle: 242
-- id: hospitalisation_count
 |- subtitle: 159
-- id: hospitalisation_lq
 |- subtitle: 247
-- id: image_detection_count
 |- subtitle: 159
-- id: image_detection_lq
 |- subtitle: 247
-- id: impact_studies_2_count
 |- subtitle: 160
-- id: impact_studies_2_lq
 |- subtitle: 248
-- id: impact_studies_3_count
 |- subtitle: 160
-- id: impact_studies_3_lq
 |- subtitle: 248
-- id: impact_studies_count
 |- subtitle: 158
-- id: impact_studies_lq
 |- subtitle: 246
-- id: indirect_impacts_count
 |- subtitle: 160
-- id: indirect_impacts_lq
 |- subtitle: 248
-- id: lockdown_social_distancing_2_count
 |- subtitle: 172
-- id: lockdown_social_distancing_2_lq
 |- subtitle: 260
 |- title: 64
-- id: lockdown_social_distancing_count
 |- subtitle: 170
-- id: lockdown_social_distancing_lq
 |- subtitle: 258
 |- title: 62
-- id: mental_health_count
 |- subtitle: 157
-- id: mental_health_lq
 |- subtitle: 245
-- id: patient_impact_count
 |- subtitle: 158
-- id: patient_impact_lq
 |- subtitle: 246
-- id: population_impact_count
 |- subtitle: 161
-- id: population_impact_lq
 |- subtitle: 249
-- id: prediction_count
 |- subtitle: 154
-- id: prediction_lq
 |- subtitle: 242
-- id: repurposed_treatments_count
 |- subtitle: 165
-- id: repurposed_treatments_lq
 |- subtitle: 253
-- id: serology_count
 |- subtitle: 152
-- id: serology_lq
 |- subtitle: 240
-- id: smoking_cancer_count
 |- subtitle: 158
-- id: smoking_cancer_lq
 |- subtitle: 246
-- id: social_media_count
 |- subtitle: 156
-- id: social_media_lq
 |- subtitle: 244
-- id: trials_count
 |- subtitle: 150
-- id: trials_lq
 |- subtitle: 238
-- id: vaccination_programmes_count
 |- subtitle: 166
-- id: vaccination_programmes_lq
 |- subtitle: 254
-- id: vaccines_count
 |- subtitle: 152
-- id: vaccines_lq
 |- subtitle: 240
-- id: variants_count
 |- subtitle: 152
-- id: variants_lq
 |- subtitle: 240
-- id: hrst_st_rcat_hrst_pc_act
 |- title: 165
-- id: hrst_st_rcat_hrst_pc_pop
 |- title: 151
-- id: hrst_st_rcat_hrst_ths
 |- title: 129
-- id: hrst_st_rcat_hrstc_pc_act
 |- title: 162
-- id: hrst_st_rcat_hrstc_pc_pop
 |- title: 148
-- id: hrst_st_rcat_hrstc_ths
 |- title: 126
-- id: hrst_st_rcat_hrste_pc_act
 |- title: 123
-- id: hrst_st_rcat_hrste_pc_pop
 |- title: 109
-- id: hrst_st_rcat_hrste_ths
 |- title: 87
-- id: hrst_st_rcat_hrsto_pc_act
 |- title: 126
-- id: hrst_st_rcat_hrsto_pc_pop
 |- title: 112
-- id: hrst_st_rcat_hrsto_ths
 |- title: 90
-- id: hrst_st_rcat_se_pc_act
 |- title: 108
-- id: hrst_st_rcat_se_pc_pop
 |- title: 94
-- id: hrst_st_rcat_se_ths
 |- title: 72
-- id: pat_ep_rtec_avi_gdp_beur
 |- title: 115
-- id: pat_ep_rtec_avi_nr
 |- title: 94
-- id: pat_ep_rtec_avi_p_mact
 |- title: 133
-- id: pat_ep_rtec_avi_p_mhab
 |- title: 111
-- id: pat_ep_rtec_cab_gdp_beur
 |- title: 148
-- id: pat_ep_rtec_cab_nr
 |- title: 127
-- id: pat_ep_rtec_cab_p_mact
 |- title: 166
-- id: pat_ep_rtec_cab_p_mhab
 |- title: 144
-- id: pat_ep_rtec_cte_gdp_beur
 |- title: 131
-- id: pat_ep_rtec_cte_nr
 |- title: 110
-- id: pat_ep_rtec_cte_p_mact
 |- title: 149
-- id: pat_ep_rtec_cte_p_mhab
 |- title: 127
-- id: pat_ep_rtec_ht_gdp_beur
 |- title: 124
-- id: pat_ep_rtec_ht_nr
 |- title: 103
-- id: pat_ep_rtec_ht_p_mact
 |- title: 142
-- id: pat_ep_rtec_ht_p_mhab
 |- title: 120
-- id: pat_ep_rtec_lsr_gdp_beur
 |- title: 112
-- id: pat_ep_rtec_lsr_nr
 |- title: 91
-- id: pat_ep_rtec_lsr_p_mact
 |- title: 130
-- id: pat_ep_rtec_lsr_p_mhab
 |- title: 108
-- id: pat_ep_rtec_mge_gdp_beur
 |- title: 145
-- id: pat_ep_rtec_mge_nr
 |- title: 124
-- id: pat_ep_rtec_mge_p_mact
 |- title: 163
-- id: pat_ep_rtec_mge_p_mhab
 |- title: 141
-- id: pat_ep_rtec_smc_gdp_beur
 |- title: 121
-- id: pat_ep_rtec_smc_nr
 |- title: 100
-- id: pat_ep_rtec_smc_p_mact
 |- title: 139
-- id: pat_ep_rtec_smc_p_mhab
 |- title: 117
-- id: pat_ep_rtot_gdp_beur
 |- title: 95
-- id: pat_ep_rtot_nr
 |- title: 74
-- id: pat_ep_rtot_p_mact
 |- title: 113
-- id: pat_ep_rtot_p_mhab
 |- title: 91
-- id: rd_e_gerdreg_eur_hab_bes
 |- title: 98
-- id: rd_e_gerdreg_eur_hab_gov
 |- title: 89
-- id: rd_e_gerdreg_eur_hab_hes
 |- title: 95
-- id: rd_e_gerdreg_eur_hab_pnp
 |- title: 97
-- id: rd_e_gerdreg_eur_hab_total
 |- title: 83
-- id: rd_e_gerdreg_mio_eur_bes
 |- title: 91
-- id: rd_e_gerdreg_mio_eur_gov
 |- title: 82
-- id: rd_e_gerdreg_mio_eur_hes
 |- title: 88
-- id: rd_e_gerdreg_mio_eur_pnp
 |- title: 90
-- id: rd_e_gerdreg_mio_eur_total
 |- title: 76
-- id: rd_e_gerdreg_mio_nac_bes
 |- title: 113
-- id: rd_e_gerdreg_mio_nac_gov
 |- title: 104
-- id: rd_e_gerdreg_mio_nac_hes
 |- title: 110
-- id: rd_e_gerdreg_mio_nac_pnp
 |- title: 112
-- id: rd_e_gerdreg_mio_nac_total
 |- title: 98
-- id: rd_e_gerdreg_mio_pps_bes
 |- title: 119
-- id: rd_e_gerdreg_mio_pps_gov
 |- title: 110
-- id: rd_e_gerdreg_mio_pps_hes
 |- title: 116
-- id: rd_e_gerdreg_mio_pps_kp05_bes
 |- title: 134
-- id: rd_e_gerdreg_mio_pps_kp05_gov
 |- title: 125
-- id: rd_e_gerdreg_mio_pps_kp05_hes
 |- title: 131
-- id: rd_e_gerdreg_mio_pps_kp05_pnp
 |- title: 133
-- id: rd_e_gerdreg_mio_pps_kp05_total
 |- title: 119
-- id: rd_e_gerdreg_mio_pps_pnp
 |- title: 118
-- id: rd_e_gerdreg_mio_pps_total
 |- title: 104
-- id: rd_e_gerdreg_pc_gdp_bes
 |- title: 121
-- id: rd_e_gerdreg_pc_gdp_gov
 |- title: 112
-- id: rd_e_gerdreg_pc_gdp_hes
 |- title: 118
-- id: rd_e_gerdreg_pc_gdp_pnp
 |- title: 120
-- id: rd_e_gerdreg_pc_gdp_total
 |- title: 106
-- id: rd_e_gerdreg_pps_hab_kp05_bes
 |- title: 149
-- id: rd_e_gerdreg_pps_hab_kp05_gov
 |- title: 140
-- id: rd_e_gerdreg_pps_hab_kp05_hes
 |- title: 146
-- id: rd_e_gerdreg_pps_hab_kp05_pnp
 |- title: 148
-- id: rd_e_gerdreg_pps_hab_kp05_total
 |- title: 134
-- id: rd_p_persreg_f_rse_fte_bes
 |- title: 157
-- id: rd_p_persreg_f_rse_fte_gov
 |- title: 148
-- id: rd_p_persreg_f_rse_fte_hes
 |- title: 154
-- id: rd_p_persreg_f_rse_fte_pnp
 |- title: 156
-- id: rd_p_persreg_f_rse_fte_total
 |- title: 142
-- id: rd_p_persreg_f_rse_hc_bes
 |- title: 141
-- id: rd_p_persreg_f_rse_hc_gov
 |- title: 132
-- id: rd_p_persreg_f_rse_hc_hes
 |- title: 138
-- id: rd_p_persreg_f_rse_hc_pnp
 |- title: 140
-- id: rd_p_persreg_f_rse_hc_total
 |- title: 126
-- id: rd_p_persreg_f_rse_pc_act_fte_bes
 |- title: 217
-- id: rd_p_persreg_f_rse_pc_act_fte_gov
 |- title: 208
-- id: rd_p_persreg_f_rse_pc_act_fte_hes
 |- title: 214
-- id: rd_p_persreg_f_rse_pc_act_fte_pnp
 |- title: 216
-- id: rd_p_persreg_f_rse_pc_act_fte_total
 |- title: 202
-- id: rd_p_persreg_f_rse_pc_act_hc_bes
 |- title: 206
-- id: rd_p_persreg_f_rse_pc_act_hc_gov
 |- title: 197
-- id: rd_p_persreg_f_rse_pc_act_hc_hes
 |- title: 203
-- id: rd_p_persreg_f_rse_pc_act_hc_pnp
 |- title: 205
-- id: rd_p_persreg_f_rse_pc_act_hc_total
 |- title: 191
-- id: rd_p_persreg_f_rse_pc_emp_fte_bes
 |- title: 203
-- id: rd_p_persreg_f_rse_pc_emp_fte_gov
 |- title: 194
-- id: rd_p_persreg_f_rse_pc_emp_fte_hes
 |- title: 200
-- id: rd_p_persreg_f_rse_pc_emp_fte_pnp
 |- title: 202
-- id: rd_p_persreg_f_rse_pc_emp_fte_total
 |- title: 188
-- id: rd_p_persreg_f_rse_pc_emp_hc_bes
 |- title: 192
-- id: rd_p_persreg_f_rse_pc_emp_hc_gov
 |- title: 183
-- id: rd_p_persreg_f_rse_pc_emp_hc_hes
 |- title: 189
-- id: rd_p_persreg_f_rse_pc_emp_hc_pnp
 |- title: 191
-- id: rd_p_persreg_f_rse_pc_emp_hc_total
 |- title: 177
-- id: rd_p_persreg_f_total_fte_bes
 |- title: 151
-- id: rd_p_persreg_f_total_fte_gov
 |- title: 142
-- id: rd_p_persreg_f_total_fte_hes
 |- title: 148
-- id: rd_p_persreg_f_total_fte_pnp
 |- title: 150
-- id: rd_p_persreg_f_total_fte_total
 |- title: 136
-- id: rd_p_persreg_f_total_hc_bes
 |- title: 135
-- id: rd_p_persreg_f_total_hc_gov
 |- title: 126
-- id: rd_p_persreg_f_total_hc_hes
 |- title: 132
-- id: rd_p_persreg_f_total_hc_pnp
 |- title: 134
-- id: rd_p_persreg_f_total_hc_total
 |- title: 120
-- id: rd_p_persreg_f_total_pc_act_fte_bes
 |- title: 211
-- id: rd_p_persreg_f_total_pc_act_fte_gov
 |- title: 202
-- id: rd_p_persreg_f_total_pc_act_fte_hes
 |- title: 208
-- id: rd_p_persreg_f_total_pc_act_fte_pnp
 |- title: 210
-- id: rd_p_persreg_f_total_pc_act_fte_total
 |- title: 196
-- id: rd_p_persreg_f_total_pc_act_hc_bes
 |- title: 200
-- id: rd_p_persreg_f_total_pc_act_hc_gov
 |- title: 191
-- id: rd_p_persreg_f_total_pc_act_hc_hes
 |- title: 197
-- id: rd_p_persreg_f_total_pc_act_hc_pnp
 |- title: 199
-- id: rd_p_persreg_f_total_pc_act_hc_total
 |- title: 185
-- id: rd_p_persreg_f_total_pc_emp_fte_bes
 |- title: 197
-- id: rd_p_persreg_f_total_pc_emp_fte_gov
 |- title: 188
-- id: rd_p_persreg_f_total_pc_emp_fte_hes
 |- title: 194
-- id: rd_p_persreg_f_total_pc_emp_fte_pnp
 |- title: 196
-- id: rd_p_persreg_f_total_pc_emp_fte_total
 |- title: 182
-- id: rd_p_persreg_f_total_pc_emp_hc_bes
 |- title: 186
-- id: rd_p_persreg_f_total_pc_emp_hc_gov
 |- title: 177
-- id: rd_p_persreg_f_total_pc_emp_hc_hes
 |- title: 183
-- id: rd_p_persreg_f_total_pc_emp_hc_pnp
 |- title: 185
-- id: rd_p_persreg_f_total_pc_emp_hc_total
 |- title: 171
-- id: rd_p_persreg_t_rse_fte_bes
 |- title: 155
-- id: rd_p_persreg_t_rse_fte_gov
 |- title: 146
-- id: rd_p_persreg_t_rse_fte_hes
 |- title: 152
-- id: rd_p_persreg_t_rse_fte_pnp
 |- title: 154
-- id: rd_p_persreg_t_rse_fte_total
 |- title: 140
-- id: rd_p_persreg_t_rse_hc_bes
 |- title: 139
-- id: rd_p_persreg_t_rse_hc_gov
 |- title: 130
-- id: rd_p_persreg_t_rse_hc_hes
 |- title: 136
-- id: rd_p_persreg_t_rse_hc_pnp
 |- title: 138
-- id: rd_p_persreg_t_rse_hc_total
 |- title: 124
-- id: rd_p_persreg_t_rse_pc_act_fte_bes
 |- title: 215
-- id: rd_p_persreg_t_rse_pc_act_fte_gov
 |- title: 206
-- id: rd_p_persreg_t_rse_pc_act_fte_hes
 |- title: 212
-- id: rd_p_persreg_t_rse_pc_act_fte_pnp
 |- title: 214
-- id: rd_p_persreg_t_rse_pc_act_fte_total
 |- title: 200
-- id: rd_p_persreg_t_rse_pc_act_hc_bes
 |- title: 204
-- id: rd_p_persreg_t_rse_pc_act_hc_gov
 |- title: 195
-- id: rd_p_persreg_t_rse_pc_act_hc_hes
 |- title: 201
-- id: rd_p_persreg_t_rse_pc_act_hc_pnp
 |- title: 203
-- id: rd_p_persreg_t_rse_pc_act_hc_total
 |- title: 189
-- id: rd_p_persreg_t_rse_pc_emp_fte_bes
 |- title: 201
-- id: rd_p_persreg_t_rse_pc_emp_fte_gov
 |- title: 192
-- id: rd_p_persreg_t_rse_pc_emp_fte_hes
 |- title: 198
-- id: rd_p_persreg_t_rse_pc_emp_fte_pnp
 |- title: 200
-- id: rd_p_persreg_t_rse_pc_emp_fte_total
 |- title: 186
-- id: rd_p_persreg_t_rse_pc_emp_hc_bes
 |- title: 190
-- id: rd_p_persreg_t_rse_pc_emp_hc_gov
 |- title: 181
-- id: rd_p_persreg_t_rse_pc_emp_hc_hes
 |- title: 187
-- id: rd_p_persreg_t_rse_pc_emp_hc_pnp
 |- title: 189
-- id: rd_p_persreg_t_rse_pc_emp_hc_total
 |- title: 175
-- id: rd_p_persreg_t_total_fte_bes
 |- title: 149
-- id: rd_p_persreg_t_total_fte_gov
 |- title: 140
-- id: rd_p_persreg_t_total_fte_hes
 |- title: 146
-- id: rd_p_persreg_t_total_fte_pnp
 |- title: 148
-- id: rd_p_persreg_t_total_fte_total
 |- title: 134
-- id: rd_p_persreg_t_total_hc_bes
 |- title: 133
-- id: rd_p_persreg_t_total_hc_gov
 |- title: 124
-- id: rd_p_persreg_t_total_hc_hes
 |- title: 130
-- id: rd_p_persreg_t_total_hc_pnp
 |- title: 132
-- id: rd_p_persreg_t_total_hc_total
 |- title: 118
-- id: rd_p_persreg_t_total_pc_act_fte_bes
 |- title: 209
-- id: rd_p_persreg_t_total_pc_act_fte_gov
 |- title: 200
-- id: rd_p_persreg_t_total_pc_act_fte_hes
 |- title: 206
-- id: rd_p_persreg_t_total_pc_act_fte_pnp
 |- title: 208
-- id: rd_p_persreg_t_total_pc_act_fte_total
 |- title: 194
-- id: rd_p_persreg_t_total_pc_act_hc_bes
 |- title: 198
-- id: rd_p_persreg_t_total_pc_act_hc_gov
 |- title: 189
-- id: rd_p_persreg_t_total_pc_act_hc_hes
 |- title: 195
-- id: rd_p_persreg_t_total_pc_act_hc_pnp
 |- title: 197
-- id: rd_p_persreg_t_total_pc_act_hc_total
 |- title: 183
-- id: rd_p_persreg_t_total_pc_emp_fte_bes
 |- title: 195
-- id: rd_p_persreg_t_total_pc_emp_fte_gov
 |- title: 186
-- id: rd_p_persreg_t_total_pc_emp_fte_hes
 |- title: 192
-- id: rd_p_persreg_t_total_pc_emp_fte_pnp
 |- title: 194
-- id: rd_p_persreg_t_total_pc_emp_fte_total
 |- title: 180
-- id: rd_p_persreg_t_total_pc_emp_hc_bes
 |- title: 184
-- id: rd_p_persreg_t_total_pc_emp_hc_gov
 |- title: 175
-- id: rd_p_persreg_t_total_pc_emp_hc_hes
 |- title: 181
-- id: rd_p_persreg_t_total_pc_emp_hc_pnp
 |- title: 183
-- id: rd_p_persreg_t_total_pc_emp_hc_total
 |- title: 169
```

Commit:
changes in script and indicators + schema:

Commit:
changes in script and indicators + schema:

Commit:
updated indicator and schema

Commit:
updated indicator and schema

Juan-Mateos commented:
I have...

- [ ] Added the eurostat framework to `data/aux/framework.json`
- [ ] Changed eurostat to EUROSTAT in the schema

Re: NUTS edition in use, the Eurostat website is unfortunately sparse. In a couple of cases (e.g. patents) they say that they have used the NUTS 2010 edition to classify all regions in all years. In other cases they don't provide information about the NUTS version they are using so I assumed it is also 2010. 

I have removed from the data all observations before 2010. 

Re: long title names, this is a difficult one to address. In some cases the names are long because their description includes the sector the indicator refers to, the unit of measurement and a brief description of the numerator and denominator when we have a ratio. These are generated automatically as we produce the indicators using the Eurostat data dictionaries. We currently have ca. 400 Eurostat indicators so I can't review and shorten all those titles manually.

mindrones reviewed (changes requested):
At the moment the web app won't load EUROSTAT indicators because some indicators have non-NUTS like `US`, `RU`.

We need to make sure that all indicators only contain datapoints with regions [in this json file](https://raw.githubusercontent.com/nestauk/svizzle_atlas_distro/79668ec058da88580c5b456519f6a701afab9e7b/NUTS/unifiedNuts.json), which represents the regions we're capable of visualising as of now.

mindrones committed:
added `ui/stats/unsupportedRegions.json`:
- keys: path of a CSV containing unsupported regions
- values: a list of datapoints in that indicator CSVs referring to unsupported regions

mindrones commented:
@Juan-Mateos I've rebased this branch to `dev` and added `ui/stats/unsupportedRegions.json`:
- keys: path of a CSV containing unsupported regions
- values: a list of datapoints in that indicator CSV containing unsupported regions: these are the specific rows to be fixed or filtered out for the PR to be accepted.

Commit:
Merge branch '113_eurostat' of github.com:nestauk/eurito_indicators into 113_eurostat

Commit:
changed script

Commit:
added script

Commit:
filtered indicators

Juan-Mateos commented:
- [ ] I have implemented a script that filter outs "bad" regions from the script you shared.
